### DISCUSSION TITLE
FileDoesNotExist using Livewire

### DISCUSSION THREAD

**Discussion by jozeflambrecht** [spatie/laravel-medialibrary#2987] [Q&A]
When I upload pictures using livewire and laravel-medialibrary, I get the following error:
File `C:\Users\jozef\Code\test\storage\app\livewire-tmp/UT4GPidHRIr6ygYKAxsL04S0suUanL-metad2hjLnBuZw==-.png` does not exist
This is my code:

```
$this->property->addMedia($image->getRealPath())->withCustomProperties([
                'public' => $this->public,
                'type' => $this->type,
            ])->toMediaCollection('test');
```

However the image was uploaded (it's in the correct folder and added in the media table). Someone knows why there is a call tot eh temp file?

**Comment by ettiennelouw** [comment]
Livewire uploads the file to the temp directory and then AddMedia() function seems to move the file from the livewire-tmp location to correct location that has been specified in the config file or the relevant disk. Had the same issue but I was using s3 as tmp and file storage. This code seems to have resolved the issue for me and uses the livewire-tmp folder on the s3 bucket.

```
$user
    ->addMediaFromDisk($this->avatar->getRealPath(), 's3')
    ->setName($this->avatar->getClientOriginalName())
    ->toMediaCollection('collection', 's3');
```

Hope this helps.
